### PR TITLE
Adds query string parameter support

### PR DESCRIPTION
**Pull request by jimmygle**
* Adds grunt to npm ```devDependencies```
* Adds ```getPathName()``` method that will parse the path name from fully qualified URL's and relative URI's.
* Adds ```getQueryParams()``` method that returns an object containing query parameters from a provided URL (any string containing ```?key=value[&key=value...]```
* Adds basic unit tests around the above two methods

Example of getQueryString() method response:
```
http://google.com/?my=little&tea=pot   =>

{
    "my":"little",
    "tea":"pot"
}
```

### CODE REVIEW

**Comment by mkay581** [comment]
hey @jimmygle looks good. thank you so much for doing this. I like the `getQueryParams()` method. Guess I'm not so sure if `RouteManager.getPathName()` is anymore helpful that just doing `'/' + RouteManager.getRelativeUrl();`

File: tests/route-manager-tests.js [hunk 1]
```diff
@@ -58,6 +58,24 @@ describe('Route Manager', function () {
...
+    it('should return path name from url', function () {
+        var RouteManager = require('route-manager')({config: {}});
+        var url = 'http://my-testable-url.com/my/testable/path';
+        RouteManager.start();
+        var pathName = RouteManager.getPathName(url);
+        assert.equal('/my/testable/path', pathName, 'path name parsed from full url: ' + pathName);
+        RouteManager.stop();
+    });
+
+    it('should return query params from url', function () {
```
Comment (mkay581) [inline tests/route-manager-tests.js]: amazing man! can we add just one more test for when you don't pass anything into `RouteManager.getQueryParams()`?

**Comment by jimmygle** [comment]
My thinking with the ```getPathName()``` method was to be a more static method, like the ```getQueryParams()```. Where it doesn't require the router to actually be started to return the expected value. 

I'm sure the ```getPathName()``` and the ```getRelativeUrl()``` methods could be consolidated to perform both functions, but for now I'll just remove the ```getPathName()``` function.

**Comment by mkay581** [comment]
yeah we should probably merge the two, but we can discuss that later. Thanks for adding the additional test.

**Pull request merged**